### ISSUE TITLE
the logic in SharePointEnvironmentSettingsForm_FormClosing()

### ISSUE THREAD

**Issue by leonardo-czanoch** [tombwu/Files#1]
consider the following scenario:
  expirationWindow > 1440 and
  expirationOverride > 1440 and
  backoffInterval < 30 and
  retryCount < 5
then the MessageBox can only show "Retries must be greater than 5 times" because retryCount < 5.
is that right?
I think expirationWindow > 1440 would be checked then show the messageBox and return.

**Comment by tombwu** [comment]
Seems that you are correct. Evan could correct me.

**Comment by Evan-Yang** [comment]
no! actually, there are 4 different block to validate 5 TextBox values, show 5 different message.

**Comment by leonardo-czanoch** [comment]
@Evan-Yang , you mean if all the 5 TextBox values are wrong, then there should be 5 MessageBoxes all pop up AT THE SAME TIME?

**Comment by Evan-Yang** [comment]
Yes

**Comment by tombwu** [comment]
Then user will click OK 5 times?  I think that would be unacceptable.
